django_project/api/osm_api_client.py

```diff
@@ -129,6 +130,26 @@ def changeset_tags(comment=None):
             })
         return tags
 
+    def merge_data(self, current_data, updated_data):
+        """ Merging tags from current data to updated data
+        if on updated tags is None, delete it
+
+        :param current_data: The current data from OSM
+        :type current_data: dict
+
+        :param updated_data: The updated data from HS
+        :type updated_data: dict
+
+        :return: merged tags
+        :rtype: dict
+        """
+        if int(current_data['id']) != int(updated_data['id']):
+            raise Exception(
+                "can't update the healthsite because the data doesn't match with osm")
+        current_data['tag'].update(updated_data['tag'])
+        updated_data['tag'] = {k: v for k, v in current_data['tag'].items() if v is not None}
```

Comment:
@meomancer I see you only update tags now. I assume this will not update any polygons anymore, is this right?

Reply:
hi @nixilla 
it will not update polygon by 
https://github.com/healthsites/healthsites/pull/1359/files#diff-a9f3a325171f9b3e4fae09789fc402f4R322

tested on https://master.apis.dev.openstreetmap.org/way/4305501616